--- a/src/ftpd.h~	2021-12-16 13:12:17.000000000 +0100
+++ b/src/ftpd.h	2024-02-28 12:08:03.216568955 +0100
@@ -397,15 +397,9 @@ extern int opt_a, opt_C, opt_d, opt_F, o
 #endif
 
 #ifdef WITH_TLS
-# ifndef TLS_CONFDIR
-#  define TLS_CONFDIR "/etc/ssl/private"
-# endif
-# ifndef TLS_CERTIFICATE_FILE
-#  define TLS_CERTIFICATE_FILE TLS_CONFDIR "/pure-ftpd.pem"
-# endif
-# ifndef TLS_KEY_FILE
+#  define TLS_CERTIFICATE_FILE "/etc/pure-ftpd/ssl/pure-ftpd.pem"
 #  define TLS_KEY_FILE TLS_CERTIFICATE_FILE
-# endif
+#  define TLS_DHPARAMS_FILE "/etc/pure-ftpd/ssl/dhparams.pem"
 #endif
 
 #ifndef TLS_DEFAULT_CIPHER_SUITE
