Index: src/ftpd.c
===================================================================
RCS file: /cvsroot/pureftpd/pureftpd/src/ftpd.c,v
retrieving revision 1.599
retrieving revision 1.600
diff -u -u -r1.599 -r1.600
--- src/ftpd.c	21 Jun 2007 15:05:06 -0000	1.599
+++ src/ftpd.c	21 Jun 2007 15:07:36 -0000	1.600
@@ -223,11 +223,10 @@
 	snprintf(buf, sizeof buf, "%d %s\r\n", err, line);
 	SSL_write(tls_cnx, buf, strlen(buf));
     } else
-#else
+#endif
     {
 	printf("%d %s\r\n", err, line);
     }
-#endif
     fflush(stdout);
     logfile(priority, "%s", line);
     va_end(va);
