--- ./man/pure-pw.8.org	2005-04-18 21:51:04.000000000 +0200
+++ ./man/pure-pw.8	2008-01-18 10:06:00.000000000 +0100
@@ -49,18 +49,18 @@
 all are chrooted, and they have their own home directory.
 .SH "FILES"
 .LP 
-\fI/etc/pureftpd.passwd\fP
+\fI/etc/ftp/pureftpd.passwd\fP
 .br
-\fI/etc/pureftpd.pdb\fP 
+\fI/etc/ftp/pureftpd.pdb\fP 
 .SH "ENVIRONMENT VARIABLES"
 .LP 
 \fIPURE_PASSWDFILE\fP
 If this variable is defined, this is the default value for the text password
-file. Without this variable, /etc/pureftpd.passwd is assumed.
+file. Without this variable, /etc/ftp/pureftpd.passwd is assumed.
 .br
 \fIPURE_DBFILE\fP 
 If this variable is defined, this is the default value for the PureDB password
-file. Without this variable, /etc/pureftpd.pdb is assumed.
+file. Without this variable, /etc/ftp/pureftpd.pdb is assumed.
 .SH "EXAMPLES"
 .LP 
 Please read http://www.pureftpd.org/README.Virtual\-Users
--- ./FAQ.org	2004-07-22 09:37:00.000000000 +0200
+++ ./FAQ	2008-01-18 10:03:04.000000000 +0100
@@ -316,25 +316,25 @@
 
 
 
-* Virtual users: /etc/pureftpd.pdb .
+* Virtual users: /etc/ftp/pureftpd.pdb .
 
--> I made changes to /etc/pureftpd.passwd but the server doesn't understand
+-> I made changes to /etc/ftp/pureftpd.passwd but the server doesn't understand
 them: I can't access any account I just created.
 
-The server never reads /etc/pureftpd.passwd directly. Instead, it reads
-/etc/pureftpd.pdb (or whatever file name you gave after -lpuredb:...) .
+The server never reads /etc/ftp/pureftpd.passwd directly. Instead, it reads
+/etc/ftp/pureftpd.pdb (or whatever file name you gave after -lpuredb:...) .
 
-This file is a copy of /etc/pureftpd.passwd, but in a binary format,
+This file is a copy of /etc/ftp/pureftpd.passwd, but in a binary format,
 optimized for fast lookups.
 
-After having made a manual change to /etc/pureftpd.passwd, you must rebuild
-/etc/pureftpd.pdb with the following commands:
+After having made a manual change to /etc/ftp/pureftpd.passwd, you must rebuild
+/etc/ftp/pureftpd.pdb with the following commands:
 
 pure-pw mkdb
 
 If you add/delete/modify user accounts with pure-pw useradd/usermod/userdel/
 passwd, don't forget the '-m' option to automatically rebuild
-/etc/pureftpd.pdb and not only update /etc/pureftpd.passwd .
+/etc/ftp/pureftpd.pdb and not only update /etc/ftp/pureftpd.passwd .
 
 
 
@@ -630,7 +630,7 @@
 
 Let's start the FTP server:
 
-/usr/local/sbin/pure-ftpd -lpuredb:/etc/pureftpd.pdb -H -B
+/usr/local/sbin/pure-ftpd -lpuredb:/etc/ftp/pureftpd.pdb -H -B
 
 Everything should be ok now.
 
--- ./README.Virtual-Users.org	2004-02-29 22:49:37.000000000 +0100
+++ ./README.Virtual-Users	2008-01-18 10:05:10.000000000 +0100
@@ -101,7 +101,7 @@
 servers. Use per-ip limits instead.
 
 Ok, "joe" has been created. By default, the list of virtual users is stored
-in the /etc/pureftpd.passwd file (you can of course change this with -f
+in the /etc/ftp/pureftpd.passwd file (you can of course change this with -f
 <file>) .
 
 Let's have a look at its content:
@@ -127,7 +127,7 @@
 
                       pure-pw usermod joe -n 1000 -N 10
 
-Let's have a look at /etc/pureftpd.passwd:
+Let's have a look at /etc/ftp/pureftpd.passwd:
 
 joe:$1$LX/3.F60$bYdYwsQOYIaWq.Ko.hfI3.:500:101::/home/ftpusers/joe/./::::::1000:10485760::::::
 
@@ -173,7 +173,7 @@
       ------------------------ DISPLAYING INFO ------------------------
       
 
-To review info about one user, reading the /etc/pureftpd.passwd file is ok,
+To review info about one user, reading the /etc/ftp/pureftpd.passwd file is ok,
 but it's not really human-friendly.
 
 It's why you can use "pure-pw show", whoose syntax is:
@@ -213,22 +213,22 @@
 IMPORTANT:
 
 You can add, modify and delete users with the previous commands, or by
-editing /etc/pureftpd.passwd by hand. But the FTP server won't consider the
+editing /etc/ftp/pureftpd.passwd by hand. But the FTP server won't consider the
 changes you make to that file, until you commit them.
 
 Commiting changes really means that a new file is created from
-/etc/pureftpd.passwd (or whatever file name you choose) . That new file is a
+/etc/ftp/pureftpd.passwd (or whatever file name you choose) . That new file is a
 PureDB file. It contains exactly the same info than the other file. But in
 that file, accounts are sorted and indexed for faster access, even with
 thousands of accounts. PureDB files are binary files, don't try to view them
 or your terminal will beep like hell.
 
-Let's create a PureDB file from /etc/pureftpd.passwd. The indexed file will
-be called /etc/pureftpd.pdb (as always, choose whatever name you like):
+Let's create a PureDB file from /etc/ftp/pureftpd.passwd. The indexed file will
+be called /etc/ftp/pureftpd.pdb (as always, choose whatever name you like):
 
                                 pure-pw mkdb
                         
-this reads /etc/pureftpd.passwd and creates /etc/pureftpd.pdb by default, but
+this reads /etc/ftp/pureftpd.passwd and creates /etc/ftp/pureftpd.pdb by default, but
 to read another file, add the pdb file, optionnaly followed by -f <passwd file>
 
 For instance:
@@ -244,13 +244,13 @@
 
 You can also change something to the text passwords file (add users, change
 password, delete users, etc) and automatically run 
-"pure-pw mkdb /etc/pureftpd.pdb" afterwards. To do so, just use the -m
+"pure-pw mkdb /etc/ftp/pureftpd.pdb" afterwards. To do so, just use the -m
 switch:
 
 pure-pw passwd joe -m
 
 This command will change Joe's password in pureftpd.passwd *and* commit the
-change to /etc/pureftpd.pwd .
+change to /etc/ftp/pureftpd.pwd .
 
 
   ------------------------ ENABLING VIRTUAL USERS ------------------------
@@ -270,7 +270,7 @@
 Let's run the server with automatic creation of home directories and puredb
 authentication:
 
-/usr/local/sbin/pure-ftpd -j -lpuredb:/etc/pureftpd.pdb &
+/usr/local/sbin/pure-ftpd -j -lpuredb:/etc/ftp/pureftpd.pdb &
 
 Try to 'ftp localhost' and log in as joe.
 
@@ -283,7 +283,7 @@
 
 Just run it:
 
-                    pure-pwconvert >> /etc/pureftpd.passwd
+                    pure-pwconvert >> /etc/ftp/pureftpd.passwd
 
 
 If you do it as a non-privileged user, passwords won't be filled in. If you
@@ -298,9 +298,9 @@
   
 If defined, a PURE_PASSWDFILE environment variable can set the default path
 to the pureftpd.passwd file. Without this variable, it defaults to
-/etc/pureftpd.passwd .  
+/etc/ftp/pureftpd.passwd .  
   
 If defined, a PURE_DBFILE environment variable can set the default path
 to the pureftpd.pdb file. Without this variable, it defaults to
-/etc/pureftpd.pdb .  
+/etc/ftp/pureftpd.pdb .  
   
