diff -Nur pure-ftpd-0.99.9.orig/contrib/redhat.sysconfig pure-ftpd-0.99.9/contrib/redhat.sysconfig
--- pure-ftpd-0.99.9.orig/contrib/redhat.sysconfig	Sun Sep 16 20:07:10 2001
+++ pure-ftpd-0.99.9/contrib/redhat.sysconfig	Mon Oct  8 13:07:10 2001
@@ -52,7 +52,7 @@
 # files uploaded by other anonymous users) . So that uploads have to be
 # validated by a system administrator (chown to another user) before being
 # available for download.
-# --antiwarez
+--antiwarez
 
 
 # log the pid of the server handling the request in syslog messages
