--- pure-ftpd.conf.in~	2017-04-24 16:32:45.000000000 +0200
+++ pure-ftpd.conf.in	2017-10-04 12:41:06.402833227 +0200
@@ -9,7 +9,7 @@
 # instead of command-line options, please run the
 # following command :
 #
-# @sbindir@/sbin/pure-ftpd @sysconfdir@/pure-ftpd.conf
+# @sbindir@/sbin/pure-ftpd @sysconfdir@/ftpd/pure-ftpd.conf
 #
 # Online documentation:
 # https://www.pureftpd.org/project/pure-ftpd/doc
@@ -106,23 +106,23 @@ MaxIdleTime                  15
 
 # LDAP configuration file (see README.LDAP)
 
-# LDAPConfigFile               /etc/pureftpd-ldap.conf
+# LDAPConfigFile               /etc/ftpd/pureftpd-ldap.conf
 
 
 
 # MySQL configuration file (see README.MySQL)
 
-# MySQLConfigFile              /etc/pureftpd-mysql.conf
+# MySQLConfigFile              /etc/ftpd/pureftpd-mysql.conf
 
 
 # PostgreSQL configuration file (see README.PGSQL)
 
-# PGSQLConfigFile              /etc/pureftpd-pgsql.conf
+# PGSQLConfigFile              /etc/ftpd/pureftpd-pgsql.conf
 
 
 # PureDB user database (see README.Virtual-Users)
 
-# PureDB                       /etc/pureftpd.pdb
+# PureDB                       /etc/ftpd/pureftpd.pdb
 
 
 # Path to pure-authd socket (see README.Authentication-Modules)
@@ -156,6 +156,7 @@ MaxIdleTime                  15
 
 # 'ls' recursion limits. The first argument is the maximum number of
 # files to be displayed. The second one is the max subdirectories depth.
+# If LimitRecursion is not present default is 2000 5
 
 LimitRecursion               10000 8
 
@@ -275,7 +276,7 @@ AutoRename                   no
 
 # Prevent anonymous users from uploading new files (no = upload is allowed)
 
-AnonymousCantUpload          no
+AnonymousCantUpload          yes
 
 
 
@@ -430,8 +431,8 @@ CustomerProof                yes
 # CertFile is for a cert+key bundle, CertFileAndKey for separate files.
 # Use only one of these.
 
-# CertFile                     /etc/ssl/private/pure-ftpd.pem
-# CertFileAndKey               "/etc/pure-ftpd.pem" "/etc/pure-ftpd.key"
+# CertFile                     /etc/ftpd/ssl/private/pure-ftpd.pem
+# CertFileAndKey               "/etc/ftpd/ssl/pure-ftpd.pem" "/etc/ftpd/ssl/pure-ftpd.key"
 
 
 
