diff -urN pure-ftpd-1.0.21.org/src/ftpd.c pure-ftpd-1.0.21/src/ftpd.c
--- pure-ftpd-1.0.21.org/src/ftpd.c	2006-02-21 14:14:49.000000000 +0100
+++ pure-ftpd-1.0.21/src/ftpd.c	2007-06-21 17:58:13.317607572 +0200
@@ -2662,7 +2662,7 @@
     }
 #endif
     addreply(250, MSG_DELE_SUCCESS, name);
-    logfile(LOG_NOTICE, MSG_DELE_SUCCESS, name);    
+    logfile(LOG_NOTICE, "%s: " MSG_DELE_SUCCESS, wp, name);    
     return;
     
     denied:
