commit 659d4e968a2204395c7e1757dade46f9dfc95960
Author: Frank Denis <github@pureftpd.org>
Date:   Sat Jun 27 16:11:56 2015 +0200

    DEFAULT_CERT_FILE -> TLS_CERTIFICATE_FILE

diff --git a/src/ftpd.h b/src/ftpd.h
index 039bed4..717c3da 100644
--- a/src/ftpd.h
+++ b/src/ftpd.h
@@ -418,8 +418,8 @@ extern int opt_a, opt_C, opt_d, opt_F, opt_l, opt_R;
 # ifndef TLS_CONFDIR
 #  define TLS_CONFDIR "/etc/ssl/private"
 # endif
-# ifndef DEFAULT_CERT_FILE
-#  define DEFAULT_CERT_FILE TLS_CONFDIR "/pure-ftpd.pem"
+# ifndef TLS_CERTIFICATE_FILE
+#  define TLS_CERTIFICATE_FILE TLS_CONFDIR "/pure-ftpd.pem"
 # endif
 # ifndef TLS_DHPARAMS_FILE
 #  define TLS_DHPARAMS_FILE TLS_CONFDIR "/pure-ftpd-dhparams.pem"
diff --git a/src/globals.h b/src/globals.h
index 833cfa2..e0c1d30 100644
--- a/src/globals.h
+++ b/src/globals.h
@@ -166,7 +166,7 @@ GLOBAL0(int data_protection_level);
 GLOBAL(const char *tlsciphersuite, TLS_DEFAULT_CIPHER_SUITE);
 GLOBAL0(signed char ssl_disabled);
 GLOBAL0(signed char ssl_verify_client_cert);
-GLOBAL(const char *cert_file, DEFAULT_CERT_FILE);
+GLOBAL(const char *cert_file, TLS_CERTIFICATE_FILE);
 #endif
 
 GLOBAL0(char *atomic_prefix);

commit 32bffd1ce2670279ebc9aac0386a6d223b1119b6
Author: Frank Denis <github@pureftpd.org>
Date:   Fri Jun 26 19:02:37 2015 +0200

    Add CertFile to the configuration files wrappers

diff --git a/configuration-file/pure-config.pl.in b/configuration-file/pure-config.pl.in
index 498af50..ea675c8 100644
--- a/configuration-file/pure-config.pl.in
+++ b/configuration-file/pure-config.pl.in
@@ -60,6 +60,7 @@ my %string_switch_for = (
         AltLog                  => "-O",
         PIDFile                 => "-g",
         TLSCipherSuite          => "-J",
+        CertFile                => "-2",
 );
 
 my %numeric_switch_for = (
diff --git a/configuration-file/pure-config.py.in b/configuration-file/pure-config.py.in
index 9a4943c..c458a20 100644
--- a/configuration-file/pure-config.py.in
+++ b/configuration-file/pure-config.py.in
@@ -114,6 +114,7 @@ option_tuple = (
     ["PIDFile\s+(\S+)",                    "-g", None            ],
     ["TLSCipherSuite\s+(\S+)",             "-J", None            ],
     ["PerUserLimits\s+([:0-9]+)",          "-y", None            ],
+    ["CertFile\s+(\S+)",                   "-2", None            ],
     ["TLS\s+(\d)",                         "-Y", None            ])
 
 for option in option_tuple:
diff --git a/configuration-file/pure-ftpd.conf.in b/configuration-file/pure-ftpd.conf.in
index 74d1fb1..8591906 100644
--- a/configuration-file/pure-ftpd.conf.in
+++ b/configuration-file/pure-ftpd.conf.in
@@ -434,6 +434,12 @@ CustomerProof              yes
 
 
 
+# Certificate file, for TLS
+
+# CertFile                 /etc/ssl/private/pure-ftpd.pem
+
+
+
 # Listen only to IPv4 addresses in standalone mode (ie. disable IPv6)
 # By default, both IPv4 and IPv6 are enabled.
 
